packages/grub2/config/grub.cfg

```diff
@@ -43,24 +43,30 @@ insmod gfxterm
 insmod loopback
 insmod squash4
 
+set loopdev="loop0"
+
 menuentry "${display_name}" --id cos {
   # label is kept around for backward compatibility
   set label=${active_label}
   set img=/cOS/active.img
-  loopback loop0 /$img
-  source (loop0)/etc/cos/bootargs.cfg
-  linux (loop0)$kernel $kernelcmd ${extra_cmdline} ${extra_active_cmdline}
-  initrd (loop0)$initramfs
+  loopback $loopdev /$img
+  source ($loopdev)/etc/cos/bootargs.cfg
+  linux ($loopdev)$kernel $kernelcmd ${extra_cmdline} ${extra_active_cmdline}
+  initrd ($loopdev)$initramfs
 }
 
 menuentry "${display_name} (fallback)" --id fallback {
   # label is kept around for backward compatibility
   set label=${passive_label}
   set img=/cOS/passive.img
-  loopback loop0 /$img
-  source (loop0)/etc/cos/bootargs.cfg
-  linux (loop0)$kernel $kernelcmd ${extra_cmdline} ${extra_passive_cmdline}
-  initrd (loop0)$initramfs
+  # if loopdev already exists, we've probably already tried booting once, delete it and try again.
+  if [ -e ($loopdev) ]; then
+    loopback -d $loopdev
+  fi
```

Comment:
Couldn't we move this check in out of the menu setup just below the `set loopdev="loop0"` line?

I think it should not hurt if this is executed in all cases and it would make the code smaller and cleaner.

Probably I am missing some detail though

Reply:
Hmm, it feels like grub runs only stuff in the "menuentry" when falling back.. here is a log with moving the `loopback -d ...` out of the menuentry:

 ```
 Booting `Elemental'

error: ../../grub-core/fs/fshelp.c:258:file `/boot/vmlinuz' not found.
error: ../../grub-core/loader/i386/efi/linux.c:207:you need to load the kernel
first.

Press any key to continue...

   Falling back to `Elemental'

error: ../../grub-core/disk/loopback.c:98:device name already exists.
error: ../../grub-core/fs/fshelp.c:258:file `/boot/vmlinuz' not found.
error: ../../grub-core/loader/i386/efi/linux.c:207:you need to load the kernel
first.

Press any key to continue...

   Falling back to `Elemental (fallback)'

error: ../../grub-core/disk/loopback.c:98:device name already exists.
error: ../../grub-core/fs/fshelp.c:258:file `/boot/vmlinuz' not found.
error: ../../grub-core/loader/i386/efi/linux.c:207:you need to load the kernel
first.

Press any key to continue...

```